src/browser/core_state.ts

```diff
@@ -804,25 +805,71 @@ export function getRoutingInfoByUuidFrame(uuid: string, frame: string): RoutingI
         } else {
             writeToLog(1, `unable to find openfinWindow of child of ${app.uuid}`, true);
         }
+    } for (const view of app.views) {
+        if (frame === view.name) {
+            return {
+                name: frame,
+                webContents: view.view.webContents,
+                frameRoutingId: view.view.webContents.mainFrameRoutingId,
+                mainFrameRoutingId: view.view.webContents.mainFrameRoutingId,
+                frameName: frame,
+                _options: view._options
+            };
+        }
     }
 }
 function getWinObjByWebcontentsId(webContentsId: number) {
     const win = getWinList().find(w => w.openfinWindow && w.openfinWindow.browserWindow.webContents.id === webContentsId);
-    return win.openfinWindow;
+    return win && win.openfinWindow;
+}
+let views: OfView[] = [];
+export interface OfView extends Identity {
+    name: string;
+    view: BrowserView;
+    frames: Map<string, Shapes.ChildFrameInfo>;
+    _options: Shapes.WebOptions;
+}
+export function addBrowserView (opts: BrowserViewOpts, view: BrowserView) {
+    const {uuid, name} = opts;
+    const ofView = { frames: new Map(), uuid, _options: opts, name, view };
+    views.push(ofView);
+    const app = appByUuid(uuid);
+    if (app) {
+        app.views.push(ofView);
+    }
+    return ofView;
+}
+export function removeBrowserView (view: OfView) {
+    const app = appByUuid(view.uuid);
+    if (app) {
+        app.views = app.views.filter(v => v.name !== view.name);
+    }
+    views = views.filter(v => v.uuid !== view.uuid && v.name !== view.name);
+}
+export function getBrowserViewByIdentity({uuid, name}: Identity) {
+   return views.find(v => v.uuid === uuid && v.name === name);
+}
+function getBrowserViewByWebContentsId(webContentsId: number) {
+    return views.find(v => v.view.webContents.id === webContentsId);
 }
-
 export function getWindowInitialOptionSet(windowId: number): Shapes.WindowInitialOptionSet {
     const ofWin = <Shapes.OpenFinWindow>getWinObjById(windowId);
     return getOptionsFromOpenFinWindow(ofWin);
 }
 export function getWebContentsInitialOptionSet(webContentsId: number) {
     const ofWin = getWinObjByWebcontentsId(webContentsId);
-    return getOptionsFromOpenFinWindow(ofWin);
+    if (ofWin) {
+        return getOptionsFromOpenFinWindow(ofWin);
+    }
+    const bview = getBrowserViewByWebContentsId(webContentsId);
+    if (bview) {
+        return getOptionsFromOpenFinWindow(bview);
+    }
 }
 
-function getOptionsFromOpenFinWindow(ofWin: Shapes.OpenFinWindow) {
+function getOptionsFromOpenFinWindow(ofWin: Shapes.InjectableContext) {
     const options = ofWin._options;
-    const { uuid, name } = options;
+    const { uuid, name } = ofWin;
```

Comment:
What's the difference between pulling the `uuid`/`name` from `options` vs `ofWin`?

Reply:
Browser view.ouptuons doesn't have uuid/name because they're not options